<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Simon Schlumbohm" />

<meta name="date" content="2023-08-08" />

<title>HarmonizR_Vignette</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">HarmonizR_Vignette</h1>
<h4 class="author">Simon Schlumbohm</h4>
<h4 class="date">2023-08-08</h4>



<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>HarmonizR is a framework around the popular batch effect reduction
algorithms <code>ComBat</code> and <code>limma</code>, who, on their
own, are unable to deal with missing data points within the data.
Missing values are not uncommon in biological data, especially in the
field of proteomics, for which the tool has originally been developed.
HarmonizR uses a matrix dissection approach to circumvent the
problematic existence of missing values and still apply established
batch effect reduction strategies. Recent updates to the HarmonizR
algorithm include but are not limited to increased computational
efficiency and more reliable rescue of features (e.g. proteins/genes).
For a full overview we would like to direct the reader to our paper
published in 2022 in Nature Communications: <a href="https://doi.org/10.1038/s41467-022-31007-x" class="uri">https://doi.org/10.1038/s41467-022-31007-x</a></p>
<p>All information regarding the upcoming sections can also be found
within our SOP on Github, which is found under the following link: <a href="https://www.github.com/HSU-HPC/HarmonizR/blob/main/inst/HarmonizR_SOP.pdf" class="uri">https://www.github.com/HSU-HPC/HarmonizR/blob/main/inst/HarmonizR_SOP.pdf</a>.</p>
<p>Please have a look over there as well.</p>
</div>
<div id="installation" class="section level2">
<h2>Installation</h2>
<p>The HarmonizR implementation is 100 % written in the programming
language <code>R</code>. The easiest way to install it is using the
package <code>devtools</code> that can be installed from CRAN via
<code>install.packages(&quot;devtools&quot;)</code> while in the <code>R</code>
environment. For further information, please refer to the
<code>devtools</code> documentation.</p>
<p>Installation from Github Repository:</p>
<p>The HarmonizR package (<a href="https://github.com/HSU-HPC/HarmonizR" class="uri">https://github.com/HSU-HPC/HarmonizR</a>) leads to the
package as well as example data) can be installed directly from GitHub
via the command
<code>devtools::install_github(&quot;HSU-HPC/HarmonizR&quot;)</code>, while in the
<code>R</code> software environment.</p>
<p>Installation from HarmonizR.zip file:</p>
<p>Please make sure to have <code>devtools</code> installed. Download
the code via the green Code button (<a href="https://github.com/HSU-HPC/HarmonizR" class="uri">https://github.com/HSU-HPC/HarmonizR</a>). Unzip the
downloaded .zip file. The HarmonizR package is the folder called
HarmonizR, which was within the .zip file. While in the <code>R</code>
environment and while in the HarmonizR directory, enter in the command
line: <code>devtools::install()</code> to install the package.</p>
<p>Upon availability, HarmonizR may also be installed via Bioconductor.
For this, please run the following:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;BiocManager&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;HarmonizR&quot;</span>)</span></code></pre></div>
<p>Once installed, the HarmonizR package can be used via:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(HarmonizR)</span></code></pre></div>
</div>
<div id="example-usage" class="section level2">
<h2>Example Usage</h2>
<p>For this example, we create a simple dataframe containing 3 features
and 6 samples:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># create a dataframe with 3 rows and 6 columns filled with random numbers</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">matrix</span>(<span class="fu">rnorm</span>(<span class="at">n =</span> <span class="dv">3</span><span class="sc">*</span><span class="dv">6</span>), <span class="at">ncol =</span> <span class="dv">6</span>))</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="co"># set the column names</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(df) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;A&quot;</span>, <span class="st">&quot;B&quot;</span>, <span class="st">&quot;C&quot;</span>, <span class="st">&quot;D&quot;</span>, <span class="st">&quot;E&quot;</span>, <span class="st">&quot;F&quot;</span>)</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="co"># create a vector of row names</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>row_names <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;F1&quot;</span>, <span class="st">&quot;F2&quot;</span>, <span class="st">&quot;F3&quot;</span>)</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="co"># set the row names</span></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(df) <span class="ot">&lt;-</span> row_names</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a><span class="co"># this is what it looks like:</span></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>df</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;            A          B          C           D          E          F</span></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; F1 0.3236209  1.2189569 -0.6984405  1.25036070 -0.2652627 1.69943050</span></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; F2 1.3913068 -0.4903159  2.6637027 -0.09239651  1.3849158 0.03995239</span></span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; F3 1.3078707 -0.9493917 -0.0851510  0.37461521 -0.1313137 1.46581487</span></span></code></pre></div>
<p>Now we create a fitting description, which assigns 2 samples to a
batch (3 batches total):</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># create a vector of batch numbers</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>batch <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="at">each =</span> <span class="dv">2</span>)</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="co"># create a dataframe with 6 rows and 3 columns</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>des <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">ID =</span> <span class="fu">colnames</span>(df), <span class="at">sample =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>, <span class="at">batch =</span> batch)</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="co"># this is what it looks like:</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>des</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ID sample batch</span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1  A      1     1</span></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2  B      2     1</span></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3  C      3     2</span></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4  D      4     2</span></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5  E      5     3</span></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6  F      6     3</span></span></code></pre></div>
<p>HarmonizR usage requires a single function call of the
<code>harmonizR()</code> function. Here, we pass the created dataframes
directly for a sequential run of HarmonizR. Alternatively, a path to the
respective input files can be passed.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># use the harmonizR() function; turning off creation of an output .tsv file</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>result <span class="ot">&lt;-</span> <span class="fu">harmonizR</span>(df, des, <span class="at">output_file =</span> <span class="cn">FALSE</span>, <span class="at">cores =</span> <span class="dv">1</span>)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Initializing HarmonizR...</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Reading the files...</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Preparing...</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Splitting the data using ComBat adjustment...</span></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Rebuilding...</span></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Termination.</span></span></code></pre></div>
<p>The result mirrors the input <code>data.frame</code> closely, yet the
batch-effect has been reduced by either <code>ComBat</code> or
<code>limma</code>:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>result</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;            A           B           C         D          E          F</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; F1 0.4258245  1.14506818 -0.27705405 1.1286900 -0.1544092 1.32703920</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; F2 1.3612542 -0.08398576  2.07861649 0.1637435  1.1530552 0.09066174</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; F3 1.0845905 -0.50779942  0.05176235 0.4219263 -0.1326639 1.09230576</span></span></code></pre></div>
<p>Alternatively, S4 data may be used as input. In this case, no
description file is needed as long as a batch description is included
within the <code>SummarizedExperiment</code>. An example input may look
like this:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>nrows <span class="ot">&lt;-</span> <span class="dv">20</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>ncols <span class="ot">&lt;-</span> <span class="dv">8</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>counts <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">runif</span>(nrows <span class="sc">*</span> ncols, <span class="dv">1</span>, <span class="fl">1e4</span>), nrows)</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>colData <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Batch=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>SummExp <span class="ot">=</span> SummarizedExperiment<span class="sc">::</span><span class="fu">SummarizedExperiment</span>(</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">assays=</span><span class="fu">list</span>(<span class="at">counts=</span>counts), </span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">colData=</span>colData)</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: Multiple Methodentabellen gefunden für &#39;aperm&#39;</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: vorhergehender Import &#39;BiocGenerics::aperm&#39; durch</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; &#39;DelayedArray::aperm&#39; während des Ladens von &#39;SummarizedExperiment&#39; ersetzt</span></span></code></pre></div>
<p>It may be passed instead of <code>data_as_input</code> (following
parameter explanation).</p>
<div id="parameters" class="section level3">
<h3>Parameters</h3>
<p>HarmonizR expects 2 mandatory and a total of nine optional arguments.
First, the mandatory ones:</p>
<ul>
<li><p><code>data_as_input</code></p></li>
<li><p><code>description_as_input</code></p></li>
</ul>
<p>The first argument <code>data_as_input</code>, is the path to the raw
data, the second argument, <code>description_as_input</code>, is the
path to the description file. Both input files can be given via their
file path and do not have to be read in separately by hand. This method
is recommended, as it ensures correct operation if the notes regarding
Input above are followed. Alternatively, both parameters can be passed
as dataframes or matrices as long as they are fitting the expected input
layout.</p>
<p>Next will be four optional arguments found also within the already
published HarmonizR:</p>
<ul>
<li><p><code>algorithm</code></p></li>
<li><p><code>ComBat_mode</code></p></li>
<li><p><code>plot</code></p></li>
<li><p><code>output_file</code></p></li>
</ul>
<p>The first optional argument is the algorithm of choice.
<code>ComBat</code> will be used by default, but using the parameter
<code>algorithm</code>, either <code>&quot;ComBat&quot;</code> or
<code>&quot;limma&quot;</code> can be chosen for data adjustment.
<code>&quot;ComBat&quot;</code> serves as the default. The second optional
argument is <code>ComBat_mode</code>. This parameter is only valid once
<code>ComBat</code> is chosen for the adjustment.</p>
<p>The <code>ComBat</code> mode is abbreviated for simplicity by using
integers:</p>
<pre><code>ComBat_mode     Corresponding ComBat Arguments

1 (default)     par.prior = TRUE, mean.only = FALSE

2               par.prior = TRUE, mean.only = TRUE

3               par.prior = FALSE, mean.only = FALSE

4               par.prior = FALSE, mean.only = TRUE</code></pre>
<p>Please refer to the <code>ComBat</code> documentation for further
details.</p>
<p>The third optional parameter is <code>plot</code>. <code>plot</code>
can be set to either <code>&quot;samplemeans&quot;</code>,
<code>&quot;featuremeans&quot;</code> or <code>&quot;CV&quot;</code> and will show a boxplot
with a box for each batch depicting the chosen method. This plot will
also be saved to a .pdf. This will be either the mean for all samples,
the mean for all features or the coefficient of variation. There will be
a separate plot for the original, unaltered input dataset and for the
<code>ComBat</code>/<code>limma</code> adjusted dataset. By default,
this parameter is turned off. Since a log transformation is assumed,
this will also be accounted for before plotting. Trying to plot data
that has not been log transformed prior may lead to an unplottable
result.</p>
<p>The fourth optional parameter is <code>output_file</code>. Setting
this parameter will grant the user the ability to choose the name of
their output .tsv file. Also, a path can be set. A string is expected as
input. This parameter will default to <code>&quot;cured_data&quot;</code>,
yielding a file called <code>cured_data.tsv</code>.</p>
<p>Further, the five new parameters will be explained one-by-one:</p>
<ul>
<li><code>sort</code></li>
</ul>
<p><code>sort</code> takes one of three available sorting algorithms as
input. Either <code>&quot;sparcity_sort&quot;</code>, for a sparcity-based
sorting, <code>&quot;seriation_sort&quot;</code>, using the <code>seriation</code>
package and <code>&quot;jaccard_sort&quot;</code>, using a Jaccard-index-based
sorting approach. Sorting happens prior to the adjustment and may change
the way blocking is executed on the data. Sorting does not yield any
benefit when the <code>block</code> parameter is unused.</p>
<ul>
<li><code>block</code></li>
</ul>
<p><code>block</code> takes an integer as input which dictates, how many
batches should be packed together during matrix dissection. This
parameter may greatly reduce the amount of sub-dataframes produced and
therefore decrease the algorithm’s runtime.</p>
<ul>
<li><code>verbosity</code></li>
</ul>
<p><code>verbosity</code> takes an integer with the lowest accepted
integer being <code>0</code>. The higher the number, the more feedback
will the HarmonizR provide in the command line. A verbosity of
<code>1</code> is the default and should be sufficient. <code>0</code>
or <code>&quot;mute&quot;</code> will prevent HarmonizR from showing anything in
the command line.</p>
<ul>
<li><code>cores</code></li>
</ul>
<p><code>cores</code> gives the user the ability to control the number
of cores used by their machine during HarmonizR execution. By default,
all available cores will be used.</p>
<ul>
<li><code>ur</code></li>
</ul>
<p><code>ur</code>, short for unique removal, can be set to either
<code>TRUE</code> or <code>FALSE</code> to toggle the newly implemented
removal of unique combinations for increased feature rescue. By default,
this feature is applied, and it is strongly suggested to not turn it
off. The parameter is available for result reproducibility.</p>
</div>
</div>
<div id="session-information" class="section level2">
<h2>Session Information</h2>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; R version 4.2.2 (2022-10-31 ucrt)</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Platform: x86_64-w64-mingw32/x64 (64-bit)</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Running under: Windows 10 x64 (build 19045)</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Matrix products: default</span></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; locale:</span></span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] LC_COLLATE=German_Germany.utf8  LC_CTYPE=German_Germany.utf8   </span></span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [3] LC_MONETARY=German_Germany.utf8 LC_NUMERIC=C                   </span></span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [5] LC_TIME=German_Germany.utf8    </span></span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; attached base packages:</span></span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; other attached packages:</span></span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] HarmonizR_0.99.1</span></span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span id="cb9-20"><a href="#cb9-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [1] Rcpp_1.0.10                 bslib_0.4.2                </span></span>
<span id="cb9-21"><a href="#cb9-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [3] compiler_4.2.2              jquerylib_0.1.4            </span></span>
<span id="cb9-22"><a href="#cb9-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [5] plyr_1.8.8                  GenomeInfoDb_1.34.9        </span></span>
<span id="cb9-23"><a href="#cb9-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [7] XVector_0.38.0              MatrixGenerics_1.8.1       </span></span>
<span id="cb9-24"><a href="#cb9-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [9] bitops_1.0-7                iterators_1.0.14           </span></span>
<span id="cb9-25"><a href="#cb9-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [11] tools_4.2.2                 zlibbioc_1.44.0            </span></span>
<span id="cb9-26"><a href="#cb9-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [13] digest_0.6.31               lattice_0.20-45            </span></span>
<span id="cb9-27"><a href="#cb9-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [15] jsonlite_1.8.5              evaluate_0.21              </span></span>
<span id="cb9-28"><a href="#cb9-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [17] rlang_1.1.1                 Matrix_1.5-1               </span></span>
<span id="cb9-29"><a href="#cb9-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [19] foreach_1.5.2               DelayedArray_0.22.0        </span></span>
<span id="cb9-30"><a href="#cb9-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [21] cli_3.6.1                   rstudioapi_0.14            </span></span>
<span id="cb9-31"><a href="#cb9-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [23] yaml_2.3.7                  parallel_4.2.2             </span></span>
<span id="cb9-32"><a href="#cb9-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [25] xfun_0.39                   fastmap_1.1.1              </span></span>
<span id="cb9-33"><a href="#cb9-33" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [27] GenomeInfoDbData_1.2.9      knitr_1.43                 </span></span>
<span id="cb9-34"><a href="#cb9-34" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [29] sass_0.4.6                  S4Vectors_0.36.2           </span></span>
<span id="cb9-35"><a href="#cb9-35" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [31] IRanges_2.32.0              grid_4.2.2                 </span></span>
<span id="cb9-36"><a href="#cb9-36" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [33] stats4_4.2.2                Biobase_2.58.0             </span></span>
<span id="cb9-37"><a href="#cb9-37" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [35] R6_2.5.1                    rmarkdown_2.22             </span></span>
<span id="cb9-38"><a href="#cb9-38" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [37] codetools_0.2-18            htmltools_0.5.5            </span></span>
<span id="cb9-39"><a href="#cb9-39" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [39] matrixStats_1.0.0           BiocGenerics_0.44.0        </span></span>
<span id="cb9-40"><a href="#cb9-40" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [41] GenomicRanges_1.48.0        SummarizedExperiment_1.26.1</span></span>
<span id="cb9-41"><a href="#cb9-41" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [43] RCurl_1.98-1.12             doParallel_1.0.17          </span></span>
<span id="cb9-42"><a href="#cb9-42" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [45] cachem_1.0.8</span></span></code></pre></div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
